api/views.py

```diff
@@ -24,15 +24,20 @@
 )
 
 
-class RelayAddressViewSet(viewsets.ModelViewSet):
+class SaveToRequestUser:
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+
+class RelayAddressViewSet(SaveToRequestUser, viewsets.ModelViewSet):
     serializer_class = RelayAddressSerializer
     permissions_classes = [permissions.IsAuthenticated, IsOwner]
 
     def get_queryset(self):
         return RelayAddress.objects.filter(user=self.request.user)
 
 
-class DomainAddressViewSet(viewsets.ModelViewSet):
+class DomainAddressViewSet(SaveToRequestUser, viewsets.ModelViewSet):
```

Comment:
Permissions is ran before any other view code is ran so this should be fine. FYI, the create for RelayAddress does not go through the same sets of check as the `make_relay_address` function so I do not recommend using this for add-on users since it won't be checking the user limit, bad words on the local portion, or  if the alias was generated before but is now deleted.